### PR TITLE
fix(build): bump `@octokit/tsconfig` major version

### PR DESCRIPTION
**Pull request by kfcampbell**
This PR uses [auth-token.js#345](https://github.com/octokit/auth-token.js/pull/345) as a model to update the major octokit/tsconfig version, revert the breaking engines change (that will take place during the upcoming Node 14/16 deprecation), and adds a test-specific tsconfig.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,14 @@
   },
   "jest": {
     "preset": "ts-jest",
```
Comment (wolfy1339) [inline package.json:37]:
```suggestion
```

File: package-lock.json [hunk 2]
```diff
@@ -1,4597 +1,8 @@
 {
   "name": "@octokit/oauth-authorization-url",
   "version": "0.0.0-development",
-  "lockfileVersion": 2,
```
Comment (wolfy1339) [inline package-lock.json:4]: Looks like you went with a version of npm that's a little too old here

Comment (kfcampbell) [reply package-lock.json:4]: I'll update; one moment!

Comment (kfcampbell) [reply package-lock.json:4]: I've regenerated it with Node v18. Please let me know if you'd like to see other changes!

Comment (wolfy1339) [reply package-lock.json:4]: :sweat_smile: That is too new. If you set `lockfileVersion` to 2 manually and regenerate, it should be good

Comment (kfcampbell) [reply package-lock.json:4]: Are you sure? I believe this repo has already dropped support for Node v16 and v16 [here](https://github.com/octokit/oauth-authorization-url.js/commit/3a443adc2c78810c9817e77c0f4430941539aa0f).

Comment (wolfy1339) [reply package-lock.json:4]: Actually, yeah it should be fine

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**